src/lib/MongoQueue.ts

```diff
@@ -0,0 +1,94 @@
+import { Model, Schema } from 'mongoose';
+import { Queue, Media, QueueManager } from 'musicca';
+
+export interface IQueue {
+  _id: string;
+  list: Media[];
+}
+
+export const schema = new Schema<IQueue>({
+  _id: String,
+  list: {
+    type: [Object],
+    default: []
+  }
+});
+
+export default class MongoQueue extends Queue {
+  public db: Model<IQueue>;
+
+  public id: string;
+
+  constructor(manager: QueueManager<Queue>, id: string, { model }: { model: Model<IQueue> }) {
+    super(manager, id);
+    this.id = id;
+    this.db = model;
+    this.connect(model);
+  }
+
+  private async connect(model: Model<IQueue>) {
+    const doc = await this.db.findById(this.id);
+    // eslint-disable-next-line new-cap
+    const mod = new model({
+      _id: this.id
+    });
+    mod.isNew = !doc;
+    mod.markModified('list');
+    await mod.save();
+  }
```

Comment:
I'm not sure what is this method for since it doesn't do anything basically. Try to use [`Model.exists()`](https://mongoosejs.com/docs/api/model.html#model_Model.exists) if you're checking if a document already exists.

Reply:
If you're going to clear the list when it already exists, maybe add an option for that.

Reply:
One feature request, can you add to the option for inserting default data? If you're going to add this, then my first comment is redundant.

Example:
```ts
constructor(manager: QueueManager<Queue>, id: string, { model, data }: { model: Model<IQueue>, data: Record<string, any> }) {
  super(manager, id);
  this.id = id;
  this.db = model;
  this.connect(model, data);
}

private async connect(model: Model<IQueue>, data: Record<string, any> = {}) {
  const doc = await this.db.findById(this.id);
  // eslint-disable-next-line new-cap
  const mod = new model({
    _id: this.id,
    ...data
  });
  mod.isNew = !doc;
  mod.markModified('list');
  await mod.save();
}
```